terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -91,12 +91,16 @@ def build_provider_dependency(file, name, details = {})
 
         source_address = details.fetch("source", nil)
         version_req = details["version"]&.strip
-        hostname, namespace, name = provider_source_from(source_address, name)
-        dependency_name = source_address ? "#{namespace}/#{name}" : name
+        hostname, file_namespace, name = provider_source_from(source_address, name)
+        # NOTE: terraform-providers/name is a shorthard for the provider name
+        # matching the namespace, we separate this from the file namespace
+        # needed to find the provider in the parsed file
+        dependency_namespace = file_namespace == "terraform-providers" ? name : file_namespace
```

Comment:
Hm this isn't right, I think we need to make a request to get the correct namespace, `terraform-providers/postgresql` does not map to `postgresql/postgresql`.